src/devices/Bmi088/Bmi088.cs

```diff
@@ -0,0 +1,301 @@
+﻿using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Numerics;
+
+namespace Iot.Device.Bmi088Device
+{
+    /// <summary>
+    /// Based on the C++ implementation on https://github.com/Seeed-Studio/Grove_6Axis_Accelerometer_And_Gyroscope_BMI088
+    /// Product page: https://www.bosch-sensortec.com/products/motion-sensors/imus/bmi088.html
+    /// Data sheet: https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bmi088-ds001.pdf
+    /// </summary>
+    public class Bmi088 : IDisposable
+    {
+        /// <summary>
+        /// Default i2c address for the accelerometer
+        /// </summary>
+        public const byte DEFAULT_ACCELEROMETER_ADDRESS = 0x19;
+
+        /// <summary>
+        /// Default i2c address for the gyroscope
+        /// </summary>
+        public const byte DEFAULT_GYROSCOPE_ADDRESS = 0x69;
+
+        private const byte ReadMask = 0x80;
+
+        private readonly I2cDevice _i2cAccelerometer;
+        private readonly I2cDevice _i2cGyroscope;
+
+        private float _accelerometerMultiplier;
+        private float _gyroscopeMultiplier;
+        private AccelerometerScaleRange _accelerometerScaleRange;
+        private GyroscopeScaleRange _gyroscopeScaleRange;
+        private AccelerometerPowerMode _accelerometerPowerMode;
+        private GyroscopePowerMode _gyroscopePowerMode;
+        private AccelerometerOutputDataRate _accelerometerOutputDataRate;
+        private GyroscopeOutputDataRate _gyroscopeOutputDataRate;
+
+        /// <summary>
+        /// Initialize on default bus (1) with default I2C addresses.
+        /// </summary>
+        public Bmi088()
```

Comment:
Consider doing following:
- create new classes:
  - Bmi088AccelerometerAndThermometer
  - Bmi088Gyroscope
- make each one of them have public `DefaultI2cAddress` const field
- make it possible to change I2cAddress (so ctor should be like `Bmi088Accelerometer(I2cDevice i2cDevice = null, bool shouldDispose = false)` and on null you should: create I2cDevice with default address, enforce disposal ([similar to what we do with GpioController](https://github.com/dotnet/iot/blob/master/src/devices/README.md#gpiocontroller))
- this implementation (Bmi088) should include capabilities of all 3 classes but use implementation provided by the split classes

see example for https://github.com/dotnet/iot/blob/master/src/devices/SenseHat/SenseHat.cs to see how can this be done

also lots of time when device has two i2c addresses it may consist of two separate sensors which may have different names, if you find that names then please create a new binding out of that and make this binding use it, if not then leave this in this folder (you might need to check spec or perhaps some forums - possibly we even already support it - you might i.e. compare registers of some accelerometers we have to see if there is overlap)

If you don't find names of the split devices treat this feedback as optional although still consider for clarity.

Reply:
@Ellerbach as a side note this is the first time I'm seeing that I2cBus class would be useful for i.e. Ft4222 (you could create constructor which takes bus but leaves default addresses alone)

Reply:
Quickly looking at 16-bit accelerometers bosch provides actually each one of them might work with this code, I suspect same will be with gyroscope. I.e. https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bma490l-ds000.pdf (I know at least couple of registers match here) - would be nice to support whole family of them rather than just this one sensor

Reply:
I suspect this one as well will work here: https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BMA456-DS000.pdf

Reply:
> @Ellerbach as a side note this is the first time I'm seeing that I2cBus class would be useful for i.e. Ft4222 (you could create constructor which takes bus but leaves default addresses alone)

I It will work with multiple I2C. I tested this scenario. The I2cDevices just need to be created separately. And there is a constructor to pass both I2cDevices.